x-pack/plugins/security_solution/public/management/components/endpoint_console/status_action.tsx

```diff
@@ -0,0 +1,179 @@
+ * 2.0.
+ */
+
+import React, { memo, useEffect, useMemo } from 'react';
+import { EuiCallOut, EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import type { HttpFetchError } from '@kbn/core/public';
+import { v4 as uuidV4 } from 'uuid';
+import type { HostInfo, PendingActionsResponse } from '../../../../common/endpoint/types';
+import type { EndpointCommandDefinitionMeta } from './types';
+import { EndpointHostIsolationStatusProps } from '../../../common/components/endpoint/host_isolation';
+import { useGetEndpointPendingActionsSummary } from '../../hooks/endpoint/use_get_endpoint_pending_actions_summary';
+import { FormattedDate } from '../../../common/components/formatted_date';
+import { EndpointAppliedPolicyStatus } from '../endpoint_applied_policy_status';
+import { EndpointAgentAndIsolationStatus } from '../endpoint_agent_and_isolation_status';
+import { useGetEndpointDetails } from '../../hooks';
+import type { CommandExecutionComponentProps } from '../console/types';
+import { FormattedError } from '../formatted_error';
+
+export const EndpointStatusActionResult = memo<
+  CommandExecutionComponentProps<
+    {
+      apiCalled?: boolean;
+      endpointDetails?: HostInfo;
+      detailsFetchError?: HttpFetchError;
+      endpointPendingActions?: PendingActionsResponse;
+    },
+    EndpointCommandDefinitionMeta
+  >
+>(({ command, status, setStatus, store, setStore }) => {
+  const endpointId = command.commandDefinition?.meta?.endpointId as string;
+  const { endpointPendingActions, endpointDetails, detailsFetchError, apiCalled } = store;
+  const isPending = status === 'pending';
+
+  const {
+    isFetching,
+    isFetched,
+    refetch: fetchEndpointDetails,
+  } = useGetEndpointDetails(endpointId, { enabled: false });
+
+  const { refetch: fetchEndpointPendingActionsSummary } = useGetEndpointPendingActionsSummary(
+    [endpointId],
+    { enabled: false }
+  );
+
+  const pendingIsolationActions = useMemo<
+    Pick<Required<EndpointHostIsolationStatusProps>, 'pendingIsolate' | 'pendingUnIsolate'>
+  >(() => {
+    if (endpointPendingActions?.data.length) {
+      const pendingActions = endpointPendingActions.data[0].pending_actions;
+
+      return {
+        pendingIsolate: pendingActions.isolate ?? 0,
+        pendingUnIsolate: pendingActions.unisolate ?? 0,
+      };
+    }
+    return {
+      pendingIsolate: 0,
+      pendingUnIsolate: 0,
+    };
+  }, [endpointPendingActions?.data]);
+
+  useEffect(() => {
+    if (!apiCalled) {
+      setStore((prevState) => {
+        return {
+          ...prevState,
+          apiCalled: true,
+        };
+      });
+
+      // Using a unique `queryKey` here and below so that data is NOT updated
+      // from cache when future requests for this endpoint ID is done again.
+      fetchEndpointDetails({ queryKey: uuidV4() })
```

Comment:
I guess that wold work as well. I did it this way so that I did not have to `useMemo()` the `queryKey`. This ensures that I call `fetchEndpiontDetails()` only once.